### ISSUE TITLE
Network graph:  series.setData does not remove existing nodes.

### ISSUE THREAD

**Issue by tavinashb** [highcharts/highcharts#10163]
I am trying update network graph data. But existing nodes are not getting removed.

#### Expected behavior
Existing nodes should be removed and replaced with new nodes as per data.

#### Actual behaviour
Existing nodes are not getting removed.

#### Live demo with steps to reproduce

[https://jsfiddle.net/px964goa/1/](https://jsfiddle.net/px964goa/1/)

Click on any node. I've added `setData` function on the same.

**Comment by pawelfus** [comment]
Hi @tavinashb 

Thank you for reporting the issue! My apologies for the late reply, I didn't receive notification about this ticket from github.

You are right, it's broken. Let me fix that.

**Comment by michaelthegreat** [comment]
Do I have to wait for version 7.1.0 to have this fix? I'm having trouble pulling the branch @pawelfus  merged his fix into (feature/networkgraph-verlet) . I'm trying to npm install it into my node modules and build it but that isnt really working .

**Comment by pawelfus** [comment]
I don't recommend using work-in-progress branches, those can be unstable. Possible workaround is to modify a few of internal options to fix this: https://jsfiddle.net/BlackLabel/3n25trfk/16/

Wrapper:
```javascript
Highcharts.wrap(
  Highcharts.seriesTypes.networkgraph.prototype, 'generatePoints',
  function(p) {
    if (this.nodes) {
      this.nodes.forEach(function(node) {
        node.destroy();
      });
      this.nodes.length = 0;
    }
    return p.apply(this, Array.prototype.slice.call(arguments, 1));
  }
);
```
And before setting data:
```javascript
            chart.series[0].options.nodes = [];
            chart.series[0].layout.nodes.length = 0;
            chart.series[0].layout.links.length = 0;
            chart.series[0].setData(arr);
```

It's not a perfect workaround, might have some flaws, for example drag&drop can not be fixed without modifying sources. 

I know v7.1 doesn't have the release date but I think end of Q1/beginning of Q2 should be ready.

**Comment by michaelthegreat** [comment]
@pawelfus thank you so much

**Comment by pawelfus** [comment]
Closing since it's merged to the v7.1 branch.
